product-mini/platforms/linux-sgx/enclave-sample/Enclave/Enclave.edl

```diff
@@ -20,11 +20,71 @@ enclave {
 
     trusted {
         /* define ECALLs here. */
-        public void ecall_handle_command(unsigned cmd,
-                                         [in, out, size=cmd_buf_size]uint8_t *cmd_buf,
-                                         unsigned cmd_buf_size);
-        public void ecall_iwasm_main([user_check]uint8_t *wasm_file_buf,
-                                     uint32_t wasm_file_size);
+      public bool ecall_handle_cmd_init_runtime(uint32_t max_thread_num);
+      public void ecall_handle_cmd_destroy_runtime();
+      public bool ecall_handle_cmd_load_module(
+        [in, size=wasm_file_size] char *wasm_file,
+        uint32_t wasm_file_size,
+        [out, size=error_buf_size] char *error_buf,
+        uint32_t error_buf_size,
+        [out] uint16_t *enclave_module_idx
+      );
+      public bool ecall_handle_cmd_unload_module(uint16_t enclave_module_idx);
+      public bool ecall_handle_cmd_instantiate_module(
+        uint16_t enclave_module_idx,
+        uint32_t stack_size,
+        uint32_t heap_size,
+        [out, size=error_buf_size] char *error_buf,
+        uint32_t error_buf_size,
+        [out] uint16_t *module_inst_idx
+      );
+      public bool ecall_handle_cmd_deinstantiate_module(uint16_t wasm_module_inst_idx);
+      public bool ecall_handle_cmd_get_exception(
+        uint16_t wasm_module_inst_idx,
+        [out, size=exception_size] char *exception,
+        uint32_t exception_size
+      );
+      public bool ecall_handle_cmd_exec_app_main(
+        uint16_t wasm_module_inst_idx,
+        [in, count=app_argc] char **u_app_argv,
+        uint32_t app_argc
+      );
+      public bool ecall_handle_cmd_exec_app_func(
+        uint16_t wasm_module_inst_idx,
+        [in, string] const char *func_name,
+        [in, count=app_argc] char **u_app_argv,
+        uint32_t app_argc
+      );
+      public void ecall_handle_cmd_set_log_level(int log_level);
+      public bool ecall_handle_cmd_set_wasi_args(
+        uint16_t enclave_module_idx,
+        [in, count=dir_list_size] const char **dir_list,
+        uint32_t dir_list_size,
+        [in, count=env_list_size] const char **env_list,
+        uint32_t env_list_size,
+        int stdinfd,
+        int stdoutfd,
+        int stderrfd,
+        [in, count=wasi_argc] char **wasi_argv,
+        uint32_t wasi_argc,
+        [in, count=addr_pool_list_size] const char **addr_pool_list,
+        uint32_t addr_pool_list_size
+      );
+      public void ecall_handle_cmd_get_version(
+        [out] uint32_t *major,
+        [out] uint32_t *minor,
+        [out] uint32_t *patch
+      );
+      public uint32_t ecall_handle_cmd_get_pgo_prof_buf_size(uint16_t wasm_module_inst_idx);
```

Comment:
maybe use macro `#if WASM_ENABLE_STATIC_PGO != 0` for those two functions?

Reply:
> maybe use macro `#if WASM_ENABLE_STATIC_PGO != 0` for those two functions?

Since EDL file is processed by sgx_edger8r, it can't pass like `-DWASM_ENABLE_STATIC_PGO` to preprocess of this file, so when we add `#if WASM_ENABLE_STATIC_PGO != 0` to EDL file, we must add `#define WASM_ENABLE_STATIC_PGO 1 or 0` in EDL file. So I think maybe we can always exposure this two interfaces and add macro check in Enclave.cpp, when `WASM_ENABLE_STATIC_PGO == 0`, keep this two functions empty.

Reply:
I guess that would be okay since the command line options are controlled by macro, wouldn't call those two functions anyway when the macro is disabled.